src/builder/yara_expression_builder.cpp

```diff
@@ -716,7 +716,7 @@ YaraExpressionBuilder hexIntVal(std::uint64_t value)
 YaraExpressionBuilder doubleVal(double value)
 {
 	auto ts = std::make_shared<TokenStream>();
-	TokenIt token = ts->emplace_back(TokenType::DOUBLE, value, numToStr(value));
+	TokenIt token = ts->emplace_back(TokenType::DOUBLE, value, numToStr(value, std::dec, false, false, true));
```

Comment:
This feels a little cryptic for me.

- which of this flags is relevant for this call?
- which of this is just to satisfy defaults?

However, using this implementation I don't know how to make this better.

Reply:
Yeah, it is. I changed `numToStr` as you suggested to automatically detect floating type instead of using flag